### DISCUSSION TITLE
pg_graphql Relationship Field not working

### DISCUSSION THREAD

**Discussion by cspecter** [supabase/supabase#7667] [Questions] [answered]
I am not understanding how this works: https://supabase.github.io/pg_graphql/configuration/#relationships-field

I have a table that has two columns that reference the same table. Right now, I am only getting one of the references.

```sql
create table "files" {
     id serial primary key,
     name text
}

create table "users" {
     id serial primary key,
     name text
     profile_picture_id not null references files(id),
    banner_id not null references files(id),
}

-- Adding comments like this
comment on constraint users_profile_picture_id_fkey on users is E'@graphql({"foreign_name": "files", "local_name": "profile_picture"})';

comment on constraint users_banner_id_fkey on users is E'@graphql({"foreign_name": "files", "local_name": "banner"})';

```

I see that the constraints  `users_profile_picture_id_fkey` and users_banner_id_fkey exists, but the GraphQl schema is not giving me the `banner` and `profile_picture` files. I don't understand what "foreign_key" and "local_key" refer to.

Also this breaks completely if I set `comment on schema public is e'@graphql({"inflect_names": true})';`

Update**

For example. Looking at [this test](https://github.com/supabase/pg_graphql/blob/be162c5613cbb18d3f4796f079819e2ef0e30bd6/test/sql/override_relationship_field_name.sql). `foreign_key` is referencing "author", but author is not in either of these tables.

**Comment by cspecter** [comment, marked as answer]
Ok. I answered my first question. When pg_graphql creates a relationship, it creates a key in the schema for the related table, in my example this is `files`.  With a little experimentation, this is what I figured out. 

`foreign_name`: This is what you want to rename this key to. (Definitely needs better documentation on this)
`local_name': This is the name that pg_graphql automatically gives the key. In my case it's `files`, i.e., the name you want to change.

My final comments were:
```
comment on constraint users_profile_picture_id_fkey on users is E'@graphql({"foreign_name": "profile_picture", "local_name": "files"})';

comment on constraint users_banner_id_fkey on users is E'@graphql({"foreign_name": "banner", "local_name": "files"})';
```

---

My current issue is that turning on `inflect_names` completely breaks introspection in all cases. It just 404s on introspection, so not a lot of info to go by.

**Comment by bh865** [comment]
Just popping in to say thanks on this.. helped me!
